mlir-tensorrt/compiler/lib/Conversion/StablehloToTensorRT/StablehloToTensorRT.cpp

```diff
@@ -1947,19 +1947,26 @@ struct HloSliceConverter : public ConvertHloOpToTensorRTPattern<HloOpType> {
     if (failed(startIndices))
       return rewriter.notifyMatchFailure(
           op, "could not convert i64 offsets to i32");
+    FailureOr<SmallVector<int32_t>> limitIndices =
+        truncateI64ToI32(loc, op.getLimitIndices());
+    if (failed(limitIndices))
+      return rewriter.notifyMatchFailure(
+          op, "could not convert i64 offsets to i32");
     FailureOr<SmallVector<int32_t>> strides =
         truncateI64ToI32(loc, op.getStrides());
     if (failed(strides))
       return rewriter.notifyMatchFailure(op,
                                          "could not convert i64 stride to i32");
-    FailureOr<SmallVector<int32_t>> i32Shape =
-        truncateI64ToI32(loc, op.getType().getShape());
-    if (failed(i32Shape))
-      return rewriter.notifyMatchFailure(op,
-                                         "could not convert i64 shape to i32");
+
+    // Result shape is `ceil((limit_indices - start_indices)/stride)`
+    SmallVector<int32_t> i32Shape(limitIndices->size());
+    for (size_t i = 0; i < limitIndices->size(); i++) {
+      i32Shape[i] = std::ceil((((*limitIndices)[i] - (*startIndices)[i])) /
```

Comment:
I think you mean to use `llvm::divideCeil` here?

Reply:
This is better